include/gatekeeper_net.h

```diff
@@ -24,10 +24,27 @@
 
 #include <rte_ethdev.h>
 
+#include "gatekeeper_flow.h"
+
 /* To mark whether Gatekeeper server configures IPv4 or IPv6. */
 #define GK_CONFIGURED_IPV4 (1)
 #define GK_CONFIGURED_IPV6 (2)
 
+#define IPv6_DEFAULT_VTC_FLOW   (0x60000000)
+#define IPv6_DEFAULT_HOP_LIMITS (0xFF)
+
+/* Store information about a packet. */
+struct ipacket {
+	/* Flow identifier for this packet. */
+	struct ip_flow  flow;
+	/* Pointer to the packet itself. */
+	struct rte_mbuf *pkt;
+	/* The length of the packet. */
+	uint16_t        len;
```

Comment:
True, but it requires another memory access to go through the `struct rte_mbuf`. Keeping the length in `struct ipacket` saves us that step, and is useful for later code to just refer to this field directly rather than having to call `rte_pktmbuf_data_len()` again.